cli/internal/runcache/runcache.go

```diff
@@ -162,11 +163,11 @@ func (tc TaskCache) RestoreOutputs(ctx context.Context, terminal *cli.PrefixedUi
 		terminal.Warn(ui.Dim(fmt.Sprintf("Failed to check if we can skip restoring outputs for %v: %v. Proceeding to check cache", tc.pt.TaskID, err)))
 		changedOutputGlobs = tc.repoRelativeGlobs
 	}
-	hasChangedOutputs := len(changedOutputGlobs) > 0
+	hasChangedOutputs := len(changedOutputGlobs.Inclusions)+len(changedOutputGlobs.Exclusions) > 0
```

Comment:
Hmm so reading the `GlobWatcher` code a little more, should we extend that as well to be aware of exclusions? Because something like `OnFileWatchEvent` seems to check if a file matches a glob, but it doesn't use exclusions at all.